### PR TITLE
StringExtensions: Refactor into partial classes and optimize methods

### PR DESCRIPTION
**Pull request by AndyButland**
## Description

This PR picks up https://github.com/umbraco/Umbraco-CMS/pull/21080 that the contributor provided with some useful file clean-up and optimizations, but due to conflicts and additional files added, wasn't in a position to be merged.  With permission I've taken the main parts of it, extended a little, and applied here.

The changes look a lot from a git diff perspective, but functionally they are quite minor.

The first step - in the first commit - I've used partial classes to break up the large `StringExtensions.cs` into more manageable files (smaller, and grouped by functionality).

Then I've added or completed the XML header comments on all public extension methods.

After that each commit takes a single optimization - first by adding a detailed unit test to verify that the optimization doesn't change any behaviour, and then applying the optimization.  Where the optimization wasn't obviously going to be an improvement, I've benchmarked using a temporary unit test, and verified the benefit (e.g. `StringWhitespace`, I compared the original to a compiled regular expression and the provided implementation, to verify the latter 50%+ faster).

Finally I did make two bug fixes that were revealed in edge cases from the introduced unit tests, as listed below.

## Change Summary
- Refactored the large `StringExtensions.cs` (1600+ lines) into organized partial class files by functionality:
  - `StringExtensions.Culture.cs` - Culture/invariant comparison and case conversion
  - `StringExtensions.Encoding.cs` - Encoding, hashing, hex conversion, base64
  - `StringExtensions.Manipulation.cs` - String transformation, trimming, replacing, splitting
  - `StringExtensions.Parsing.cs` - Type conversion, GUID generation, enum parsing
  - `StringExtensions.Sanitization.cs` - Cleaning, validation, null checking, XSS prevention

- Performance optimizations
  - `IsLowerCase` / `IsUpperCase` - Eliminated string allocations by using `char.ToLowerInvariant()` directly
  - `ReplaceNonAlphanumericChars` - Changed from O(n*m) to O(n) using StringBuilder
  - `StripWhitespace` - Single-pass StringBuilder with early exit when no whitespace
  - `DecodeFromHex` - Eliminated O(n²) string concatenation with StringBuilder
  - `StripHtml` - Source-generated regex for better performance
  - `EnsureStartsWith` / `EnsureEndsWith` - Eliminated unnecessary `ToString()` allocations
  - `StripNewLines` / `ToSingleLine` - Single-pass instead of multiple `Replace()` calls

- Bug fixes:
  - `StripHtml` - Fixed overly aggressive regex that incorrectly stripped `<>` and math expressions like `5 < 10 > 3`
  - `EnsureStartsWith(string)` - Fixed inconsistent case sensitivity between `StartsWith` and `TrimStart`

- Added comprehensive unit tests for all optimized methods

## Testing
Given I've already "reviewed" to an extent by taking this from a community PR, I think it's sufficient to review the description of this PR, and verify that the changes described have been applied.  The only functional changes are those listed in the "Bug fixes" above.

### CODE REVIEW

File: src/Umbraco.Core/Extensions/StringExtensions.Manipulation.cs [hunk 1]
```diff
@@ -0,0 +1,926 @@
...
+    /// <param name="input">The string to process.</param>
+    /// <returns>The string with all carriage returns and line feeds removed.</returns>
+    public static string StripNewLines(this string input)
+    {
+        if (string.IsNullOrEmpty(input))
+        {
+            return input;
+        }
+
+        // Check if any newlines exist to avoid allocating the StringBuilder unnessarily.
```
[the remark this answers is not in the document]
Comment (AndyButland) [reply src/Umbraco.Core/Extensions/StringExtensions.Manipulation.cs]: Fixed.

File: src/Umbraco.Core/Extensions/StringExtensions.Manipulation.cs [hunk 2]
```diff
@@ -0,0 +1,926 @@
...
+    /// <param name="text">The text to convert.</param>
+    /// <returns>The text as a single line, or the original text if it is null or empty.</returns>
+    public static string ToSingleLine(this string text)
+    {
+        if (string.IsNullOrEmpty(text))
+        {
+            return text;
+        }
+
+        // Check if any newlines exist to avoid allocating the StringBuilder unnessarily.
```
[the remark this answers is not in the document]
Comment (AndyButland) [reply src/Umbraco.Core/Extensions/StringExtensions.Manipulation.cs]: Fixed.

File: src/Umbraco.Core/Extensions/StringExtensions.Parsing.cs [hunk 3]
```diff
@@ -0,0 +1,240 @@
...
+
+        // convert the name to a sequence of octets (as defined by the standard or conventions of its namespace) (step 3)
+        // ASSUME: UTF-8 encoding is always appropriate
+        var nameBytes = Encoding.UTF8.GetBytes(name);
+
+        // convert the namespace UUID to network order (step 3)
+        var namespaceBytes = namespaceId.ToByteArray();
+        SwapByteOrder(namespaceBytes);
+
+        // comput the hash of the name space ID concatenated with the name (step 4)
```
[the remark this answers is not in the document]
Comment (AndyButland) [reply src/Umbraco.Core/Extensions/StringExtensions.Parsing.cs]: Fixed.

File: src/Umbraco.Core/Extensions/StringExtensions.Encoding.cs [hunk 4]
```diff
@@ -0,0 +1,427 @@
...
+        {
+            return string.Empty;
+        }
+
+        var sb = new StringBuilder(hexValue.Length / 2);
+        for (var i = 0; i < hexValue.Length; i += 2)
+        {
+            ReadOnlySpan<char> hexPair = hexValue.AsSpan(i, 2);
+            sb.Append((char)Convert.ToUInt32(hexPair.ToString(), 16));
+        }
```
[the remark this answers is not in the document]
Comment (AndyButland) [reply src/Umbraco.Core/Extensions/StringExtensions.Encoding.cs:138]: Have applied this suggestion, throwing an `ArgumentException` if an odd length input is provided.
